pkg/kubelet/kubelet.go

```diff
@@ -1076,6 +1070,13 @@ func (kl *Kubelet) SyncPods(pods []api.BoundPod) error {
 			// syncPod() will handle this one.
 			continue
 		}
+		_, _, podAnnotations := ParsePodFullName(podFullName)
+		if source := podAnnotations[ConfigSourceAnnotationKey]; !kl.sourceReady(source) {
```

Comment:
@dchen1107  How long has it been possible to extract podAnnotations from a pod's name?  I'm wondering if this will work right if a user upgrades a kubelet on a machine which has older pods on it which use an older docker container naming convention?  

I guess as long as the source == "" and that source has not been seen, then we won't delete the pod, which is good.  Right?